docs/articles/commands/custom_context_checks.md

```diff
@@ -73,4 +77,18 @@ Then we use the check like such:
 [Command("dm")]
 [DirectMessageUsage(DirectMessageUsage.RequireDMs)]
 public async ValueTask RequireDMs(CommandContext commandContext) => await commandContext.RespondAsync("This command was executed in a DM!");
-```
\ No newline at end of file
+```
+
+## Advanced Features
+
+The classes you use to implement checks participate in dependency injection, and you can request any type you previously supplied to the service provider in a public constructor. Useful applications include, but are not limited to, logging or tracking how often a command executes.
+
+A single check class can also implement multiple checks, like so:
+
+```cs
+public class Check : IContextCheck<FirstAttribute>, IContextCheck<SecondAttribute>;
+```
+
+This means that all other code in that class can be shared between the two check methods, but this should be used with caution - it means that both checks are registered at once and cannot be removed independently of each other, and it means the same construction ceremony will run for both checks.
+
+There is no limit on how many different checks can reference the same attribute, they will all be supplied with that attribute. If you need a check that always runs, you can target `UnconditionalCheckAttribute`.
```

Comment:
Instead of "always runs", could we be a bit more specific and say something like "if you need a check that always runs, you can target `UnconditionalCheckAttribute`. These checks will be ran on every command, regardless if they're applied to it or not.